ts/ts/main.ts

```diff
@@ -495,3 +515,56 @@ async function handleDateSearch(event: Event): Promise<void> {
     }
   }
 }
+
+if (!$saveButton) throw new Error('$saveButton not found!');
+$saveButton.addEventListener('click', saveDate);
+
+function saveDate(): void {
+  if (!currentCelebration) {
+    return;
+  }
+  if (!$savePopUp) throw new Error('$savePopUp not found!');
+  if (!$holidayName) throw new Error('$holidayName not found!');
+  if (!$holidayName.textContent) {
+    return;
+  }
+  if (!$holidayDesc) throw new Error('$holidayDesc not found!');
+  if (!$holidayDesc.textContent) {
+    return;
+  }
```

Comment:
You could combine 528 and 532 together since they are both checking `textContent`. Potentially also adding the 523 `currentCelebration` check too.
`if (!currentCelebration || !$holidayName.textContent || !$holidayDesc.textContent) return`